### PR TITLE
kie-issues#227: Migrate KIE Tools VS Code Extensions from "redhat" to "kie-group" publisher (phase 1)

### LINKED ISSUE #227: Migrate KIE Tools VS Code Extensions from "redhat" to "kie-group" publisher (phase 1)
Some of the VS Code Extensions are being published under "redhat" on VS Code Marketplace. This issue is for changing those extensions (with the exception on DashBuilder) to be published under "kie-group".

This issue is the first phase of a multi-phase plan for the VS Code Extension. More details on the next phases will be shared on new issues.

After this task, some extensions will remain with the old naming convention:
- vscode-extension-dashbuilder-editor
- vscode-extension-kie-ba-bundle 
- vscode-extension-kogito-bundle

A separate task will be created for each of those when we decide what to do with them.

---

There will be duplicated Serverless Workflow extensions on KIE and Red Hat for some time. The reasoning behind this is to provide a smooth transition path for users that don’t have any bundle installed. After a short transition period, we plan to remove the Serverless Workflow extension from the Red Hat publisher, keeping only the KIE one.

---

```[tasklist]
### Acceptance criteria
- [ ] 1. Create a new Serverless Workflow Extension to be published to KIE on `packages/serverless-workflow-vscode-extension` with publisher equal to `"kie-group"`
- [ ] 2. Add a deprecation warning to the current Serverless Workflow Extension published on Red Hat (`vscode-extension-serverless-workflow-editor`).
- [x] 3. Change extensions for DMN/BPMN/PMML to be published to KIE (“kie-group”) and rename their packages
- [ ] 4. Create a new bundle on IBM’s org, pointing to the new extensions under the KIE org (Hosted on the new `IBM/bamoe-tools` repo)
- [x] 5. Change Kogito Bundle (`vscode-extension-kogito-bundle`) and KIE BA Bundle (`vscode-extension-kie-ba-bundle`) to point to the new extensions published under KIE
- [ ] 6. Change GitHub Actions and Jenkins to work with the new names.
```

- About AC 2.
  - See https://code.visualstudio.com/api/working-with-extensions/publishing-extension#deprecating-extensions
- About AC 3.
  - Delete extensions:
    - vscode-extension-backend
    - vscode-extension-red-hat-business-automation-bundle
      - The extensions under the Red Hat publisher will be released from the 0.26.x branch.
  - Rename extensions:
    - vscode-extension-bpmn-editor → bpmn-vscode-extension
    - vscode-extension-dmn-editor → dmn-vscode-extension
    - vscode-extension-pmml-editor → pmml-vscode-extension
    - vscode-extension-yard-editor → yard-vscode-extension
    - vscode-extension-pack-kogito-kie-editors  → kie-editors-dev-vscode-extension
- About AC 4.
  - This bundle will include:
    - BPMN
    - DMN
- About AC 5.
  - Kogito Bundle will include:
    - BPMN
    - DMN
    - Serverless Workflow
    - DashBuilder (this will still be under the Red Hat publisher)
  - KIE BA Bundle will include:
    - BPMN
    - DMN

### PR DESCRIPTION
**Pull request by paulovmr**
Closes https://github.com/kiegroup/kie-issues/issues/227

### CODE REVIEW

File: packages/vscode-extension-serverless-workflow-editor/README.md [hunk 1]
```diff
@@ -80,4 +83,4 @@ Some Open Source projects were vital for the development of this extension, and
 - [Mermaid](https://mermaid-js.github.io/): for workflow rendering and visualization;
 - [Apicurio](https://www.apicur.io/): for our service catalog integration;
 
-You can also checkout the [CNCF Serverless Workflow VSCode extension](https://marketplace.visualstudio.com/items?itemName=serverlessworkflow.serverless-workflow-vscode-extension) at VSCode Marketplace.
+You can also checkout the [CNCF Serverless Workflow VS Code extension](https://marketplace.visualstudio.com/items?itemName=serverlessworkflow.vscode-extension-serverless-workflow-editor) at VS Code Marketplace.
```
Comment (caponetto) [inline packages/vscode-extension-serverless-workflow-editor/README.md]: Please revert the change in the URL.

File: packages/serverless-workflow-vscode-extension/package.json [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+    "url": "https://github.com/kiegroup/kie-tools/issues"
+  },
+  "browser": "./dist/extension/extensionWeb.js",
+  "main": "dist/extension/extension.js",
+  "scripts": {
+    "build:dev": "rimraf dist && webpack --env dev",
+    "build:prod": "rimraf dist && webpack && pnpm pack:prod && pnpm test:it",
+    "compile": "webpack --env dev",
+    "lint": "run-script-if --bool \"$(build-env linters.run)\" --then \"kie-tools--eslint ./src\"",
+    "pack:prod": "vsce package --githubBranch main --no-dependencies -o ./dist/vscode_extension_serverless_workflow_editor_$npm_package_version.vsix",
```
Comment (caponetto) [inline packages/serverless-workflow-vscode-extension/package.json]: Please replace `vscode_extension_serverless_workflow_editor` with the new name.

File: .github/workflows/release_build.yml [hunk 3]
```diff
@@ -713,11 +702,10 @@ jobs:
         uses: ./.github/actions/bootstrap
         with:
           pnpm_filter_string: >-
-            -F vscode-extension-bpmn-editor...
-            -F vscode-extension-dmn-editor...
-            -F vscode-extension-pmml-editor...
+            -F bpmn-vscode-extension...
+            -F dmn-vscode-extension...
+            -F pmml-vscode-extension...
             -F vscode-extension-kogito-bundle...
```
Comment (caponetto) [inline .github/workflows/release_build.yml:708]: Please add the new serverless workflow extension to this filter.

File: README.md [hunk 4]
```diff
@@ -70,10 +70,10 @@ The KIE Tools project contains several applications. To develop each one of them
...
+1. After you've successfully built the project following the instructions above, open the `packages/kie-editors-dev-vscode-extension` folder on VS Code. Use a new VS Code window so that the `packages/kie-editors-dev-vscode-extension` folder shows up as root in the VS Code explorer.
 2. From there, you can Run the extension or the integration tests by using the `Debug` menu/section. You can also use the respective shortcuts (F5 to start debugging, for instance).
 3. **NOTE:** To run the VS Code extension in development mode, you need `webpack` and `webpack-cli` to be globally installed on NPM. Normally you can do that with `npm install -g webpack@^5.36.2 webpack-cli@^4.7.0`, but `sudo` may be required depending on your installation.
-4. **Remember!** If you make changes to any package other than `packages/vscode-extension-pack-kogito-kie-editors`, you have to manually rebuild them before relaunching the extension on VS Code.
+4. **Remember!** If you make changes to any package other than `packages/kie-editors-dev-vscode-extension`, you have to manually rebuild them before relaunching the extension on VS Code.
```
Comment (caponetto) [inline README.md:76]: Please update the section below (_VS Code Extension (Serverless Workflow Editor)_) with the new extension name.

File: packages/vscode-extension-serverless-workflow-editor/README.md (hunk 1 above)
Comment (paulovmr) [reply packages/vscode-extension-serverless-workflow-editor/README.md]: Done.

File: packages/serverless-workflow-vscode-extension/package.json (hunk 2 above)
Comment (paulovmr) [reply packages/serverless-workflow-vscode-extension/package.json]: Done.

File: .github/workflows/release_build.yml (hunk 3 above)
Comment (paulovmr) [reply .github/workflows/release_build.yml:708]: Done.

File: README.md (hunk 4 above)
Comment (paulovmr) [reply README.md:76]: Done.

**Review by caponetto — APPROVED** [review]
